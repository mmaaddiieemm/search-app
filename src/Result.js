import React from "react"; 

import PropTypes from "prop-types"; 

function FormatCommonData({commonData}) {
    return(
        <>
            <p>_id: {commonData._id} external_id: {commonData.external_id} created_at: {commonData.created_at}</p>
        </>
    ); 
}

// FormatCommonData.propTypes = {
//     commonData: PropTypes.shape({
//       _id: PropTypes.number,
//       external_id: PropTypes.string,
//       created_at: PropTypes.string,
//     })
// }; 

export default FormatCommonData; 