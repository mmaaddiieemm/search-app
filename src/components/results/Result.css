.Results {
    color: rgb(51, 51, 51); 
    font-size: 12pt; 
    max-width: 700px; 
}

.Type {
    color:darkgray; 
    font-size: 20pt; 
}