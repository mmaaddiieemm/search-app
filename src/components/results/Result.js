import React from 'react';
import { ListGroup } from 'react-bootstrap';
import FormatCommonData from './CommonData';
import './Result.css';
import DataRow from './DataRow';

//
function FormatCustomData({ data, customKeys }) {
  // this array of keys represents the general structure
  // for how to format the data output on the screen.
  return (
    <>
      {customKeys.map((row) => (<DataRow key={row} data={data} keys={row} />))}
    </>
  );
}


function Result({ data, customKeys, type }) {
  if (typeof data === 'undefined' || data === null) return <></>;
  return (
    <div className="Results">
      <ListGroup>
        <label className="Type">{data.length > 0 ? type : ''}</label>
        {data.map((i) => (
          <ListGroup.Item key={i.external_id}>
            <FormatCustomData data={i} customKeys={customKeys} />
            <FormatCommonData commonData={i} />
          </ListGroup.Item>
        ))}
      </ListGroup>
    </div>
  );
}
export default Result;
