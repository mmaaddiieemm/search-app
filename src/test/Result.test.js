import React from "react"; 
import renderer from 'react-test-renderer';
import Result from '../components/results/Result.js';

test('When undefined or null data is passed in to verifyData', () => {
    const component = renderer.create(
        <Result data={null} customKeys={['1', '2', '3']} type={'uot'} />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });