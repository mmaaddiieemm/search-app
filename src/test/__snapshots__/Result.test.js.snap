// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When undefined or null data is passed in to verifyData 1`] = `null`;
